lib/std/heap.zig

```diff
@@ -590,6 +590,17 @@ pub fn StackFallbackAllocator(comptime size: usize) type {
     };
 }
 
+/// An allocator intended as a placeholder for statically allocated data. Always fails on alloc and resize, and does nothing on free.
+pub const static_allocator = Allocator{
+    .ptr = undefined,
+    .vtable = &static_allocator_vtable,
+};
+const static_allocator_vtable = Allocator.VTable{
+    .alloc = Allocator.noAlloc,
+    .resize = Allocator.noResize,
```

Comment:
I would expect `resize` to work with this allocator - by allowing any shrinking and preventing any resize over the static buffer's size.

Reply:
This would be possible to allow, and I could write this change, but I don't see how it would be useful. The buffer would have to be read-only as it could point to static data, and I'm not sure when resizing would be useful for an immutable buffer.

Reply:
What if it's not read-only memory? Also, if it is read-only memory you can't really free it with the current implementation of `std.mem.Allocator` as it will attempt to write `0xAA` bytes (for `undefined`) in debug mode.